### ISSUE TITLE
Make a reusable `shapeshift_lib`.

### ISSUE THREAD

**Issue by pq** [google/dart-shapeshift#14]
Pretty please?

It would be great if we could reuse some of your comment analyses in the [linter](https://github.com/dart-lang/linter).

**Comment by srawlins** [comment]
Agreed. I was just thinking about this yesterday... right now I have exactly one output: Markdown.

At the very least, the tool should offer a machine-readable output... I'm in the process now of really cleaning up the code so that it's at least unit testable. At that point, the unit tests will probably define an API pretty well, for a shapeshift_lib. Hopefully in the next weeks this will be possible.

**Comment by srawlins** [comment]
Oh, and right now most of shapeshift is in the cli/ library, which requires `dart:io`. I'm refactoring it out so that most of shapeshift is in the common/ library.

**Comment by pq** [comment]
Cool.  Thanks.  It occurs to me that we may need to be really careful with dependencies if we don't want to get into version constraint hell.  Any thoughts on what the lib will depend on?

**Comment by srawlins** [comment]
I think the latest commits and updates to the README essentially close this issue. But I'll let you take a look, @pq.

**Comment by pq** [comment]
:+1: 

This will go up on pub soon?

FWIW: before that happens, I noticed the description was missing.  Here's a PR to add one:

https://github.com/google/dart-shapeshift/pull/16

Obviously, you may want to change the text as you see fit.

**Comment by srawlins** [comment]
Yeah, I'll leave this open till shapeshift gets up on pub. I want to tear out DocCoverage first.

**Comment by srawlins** [comment]
OK rock on. DocCoverage is in its own repo, and already [published to pub](https://pub.dartlang.org/packages/doc_coverage).

Once shapeshift gets up to pub. I'll close this.
